addons/website/static/src/js/widgets/ace.js

```diff
@@ -37,9 +38,27 @@ var WebsiteAceEditor = AceEditor.extend({
      */
     _saveResources: function () {
         return this._super.apply(this, arguments).then((function () {
-            this._updateHash();
-            window.location.reload();
-            return $.Deferred();
+            var defs = [];
+            var selectedView = _.findWhere(this.views, {id: this._getSelectedResource()});
+            if (this.currentType === 'xml' && !selectedView.website_id[0]) {
+                // When saving a generic view, the view will be COW'd and
+                // replace by the specific view after the reload. Thus the id in
+                // URL won't exist anymore. We need to find the specific ID.
+                var context = weContext.get();
+                defs.push(this._rpc({
+                    model: 'ir.ui.view',
+                    method: 'search_read',
+                    fields: ['id'],
+                    domain: [['key', '=', selectedView.key], ['website_id', '=', context.website_id]],
+                }).then((function (view) {
+                    if (view[0]) {
```

Comment:
Is not this always true ?

Reply:
There should always be a view returned as we just saved a generic view and we should always find the newly created COW view.
That is a just a check to avoid js crash later when reading the hash if for some reason the search_read does not find the view, in that case `view[0]` will be `undefined` as view will be `[]`.